packages/plugin-stage/sources/drivers/GitDriver.ts

```diff
@@ -8,113 +8,88 @@ const MESSAGE_MARKER = `Commit generated via \`yarn stage\``;
 const COMMIT_DEPTH = 11;
 
 async function getLastCommitHash(cwd: string) {
-  const {stdout: lastCommitStdout} = await execUtils.execvp(`git`, [`log`, `-1`, `--pretty=format:%H`], {cwd, strict: true});
-  return lastCommitStdout;
+  const {code, stdout} = await execUtils.execvp(`git`, [`log`, `-1`, `--pretty=format:%H`], {cwd});
+
+  if (code === 0) {
+    return stdout.trim();
+  } else {
+    return null;
+  }
 }
 
 async function genCommitMessage(cwd: string, changes: Array<stageUtils.FileAction>) {
-  const updates = new Map();
-  const removes = new Map();
-  const adds = new Map();
-  const createdPackages = [];
-  const removedPackages = [];
-
-  const jsonChanges = changes.filter(change => posix.basename(change.path) === "package.json");
-  for(const change of jsonChanges) {
-    const { action, path } = change;
+  const actions: Array<[stageUtils.ActionType, string]> = [];
+
+  const modifiedPkgJsonFiles = changes.filter(change => {
+    return posix.basename(change.path) === `package.json`;
+  });
+
+  for (const {action, path} of modifiedPkgJsonFiles) {
     const localPath = NodeFS.fromPortablePath(path);
     const relativePath = posix.relative(cwd, localPath);
+
     if (action === stageUtils.ActionType.MODIFY) {
       const commitHash = await getLastCommitHash(cwd)
       const {stdout: prevSource} = await execUtils.execvp(`git`, [`show`, `${commitHash}:${relativePath}`], {cwd, strict: true});
+
       const prevManifest = await Manifest.fromText(prevSource);
       const currManifest = await Manifest.fromFile(localPath);
+
       const allCurrDeps: Map<IdentHash, Descriptor> = new Map([...currManifest.dependencies, ...currManifest.devDependencies]);
       const allPrevDeps: Map<IdentHash, Descriptor> = new Map([...prevManifest.dependencies, ...prevManifest.devDependencies]);
 
-      for(const [indentHash, value] of allPrevDeps) {
+      for (const [indentHash, value] of allPrevDeps) {
         const pkgName = structUtils.stringifyIdent(value);
         const currDep = allCurrDeps.get(indentHash);
-        if (currDep) {
-          if(currDep.range !== value.range) {
-            const key = `Updates ${pkgName} to ${currDep.range}`;
-            setKeyValue(updates, key);
-          }
-        }
-        else {
-          const key = `Removes ${pkgName}`;
-          setKeyValue(removes, key);
+
+        if (currDep && currDep.range !== value.range) {
```

Comment:
if the range are the same here it will go into the else and treat it as a remove.

Reply:
True! Shouldn't have copy pasted 😅

Reply:
Fixed 👍